src/granite_orm/validators.cr

```diff
@@ -5,12 +5,23 @@ module Granite::ORM::Validators
 
   macro included
     macro inherited
+      @@validators = Array({field: Symbol, message: String, block: Proc(self, Bool)}).new
       @validators = Array({field: Symbol, message: String, block: Proc(Bool)}).new
       def validate!
       end
     end
   end
 
+  # First option: syntax support using procs
+  macro validate(message, block)
+    @@validators << {field: :base, message: {{message}}, block: {{block}}}
+  end
+
+  macro validate(field, message, block)
```

Comment:
Can these parameters be typed? When this code gets accidentally triggered by something like `validate(:name, "cant be blank", "must be present")` the compiler error messages are going to be confusing. Any help we can give the compiler will result in it providing better error feedback to devs.

Maybe crystal doesn't support this with macros, I don't remember.

Reply:
I don't think these need to be macros.  See if we can change them to `def`.